src/com/seafile/seadroid2/data/DataManager.java

```diff
@@ -143,8 +143,7 @@ public String getThumbnailLink(String repoName, String repoID, String filePath,
         } else {
             try {
                 String pathEnc = encodeThumbnailFilePath(filePath);
-                // TODO: If there is a "?" in the path, this will break
-                return account.getServer() + String.format("api2/repos/%s/thumbnail/%s?s=%s", repoID, pathEnc, size);
+                return account.getServer() + String.format("api2/repos/%s/thumbnail/?p=%s&size=%s", repoID, pathEnc, size);
```

Comment:
The hacky escaping of the path param in [DataManager.encodeThumbnailFilePath](https://github.com/haiwen/seadroid/tree/5a7531da03e2058c02b15a9ca4f21b459630d540/src/com/seafile/seadroid2/data/DataManager.java#L126) is no longer necessary. We can replace it with a call to `URLEncoder.encode()`.